youtube_dl/__init__.py

```diff
@@ -141,7 +147,22 @@ def _real_main(argv=None):
             parser.error('auto number start must be positive or 0')
     if opts.usetitle and opts.useid:
         parser.error('using title conflicts with using video ID')
-    if opts.username is not None and opts.password is None:
+    if opts.username is not None and opts.password is None and opts.password_from_keyring is True:
```

Comment:
As already said in https://github.com/rg3/youtube-dl/pull/10610#issuecomment-246117414 you must not touch the current behavior. Similarly to `--netrc` it should not work together with `-u` and `-p` and print an error.